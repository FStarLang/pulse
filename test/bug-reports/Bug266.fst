module Bug266

#lang-pulse
open Pulse

[@@expect_failure]
fn
test_intro (_:unit)
  requires emp
  ensures emp
{
  intro_exists (fun () -> pure False) ();
  admit()
}

let post () = emp
assume
val my_intro (p : prop)
  : stt_ghost unit emp_inames (pure p) post

(* Should fail! Pulse is not actually applying my_intro. *)
fn
test_my_intro (_:unit)
  requires emp
  ensures emp
{
  my_intro False;
  admit()
}
