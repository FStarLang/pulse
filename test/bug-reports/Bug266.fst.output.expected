>> Got issues: [
* Error 19 at Bug266.fst(12,2-12,14):
  - Failed to prove pure property: Prims.l_False
  - Assertion failed
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - See also Bug266.fst(12,31-12,36)

>>]
* Info at Bug266.fst(28,2-28,9):
  - Admitting continuation.
  - Current context:
      emp
  - In typing environment:
      _#118 :
        Pulse.Lib.Core.stt_ghost Prims.unit
          Pulse.Lib.Core.emp_inames
          (Pulse.Lib.Core.pure Prims.l_False)
          Bug266.post,
      uu___0#57 : Prims.unit

